java-checks/src/main/java/org/sonar/java/checks/security/ServerCertificatesCheck.java

```diff
@@ -33,9 +33,12 @@
 import org.sonar.plugins.java.api.tree.NewClassTree;
 import org.sonar.plugins.java.api.tree.ThrowStatementTree;
 import org.sonar.plugins.java.api.tree.Tree;
+import org.sonarsource.analyzer.commons.annotations.DeprecatedRuleKey;
 
-@Rule(key = "S4424")
-public class TrustManagerCertificateCheck extends IssuableSubscriptionVisitor {
+@DeprecatedRuleKey(ruleKey = "S4424", repositoryKey = "squid")
+@DeprecatedRuleKey(ruleKey = "S4424", repositoryKey = "java")
+@Rule(key = "S4830")
```

Comment:
Pay special attention to these lines.

To me, this is required to support update from both:
- version < 6.0 to 6.1 (first `DeprecatedRuleKey`)
- version > 6.0 to 6.1 (second)

See [documentation](https://github.com/SonarSource/sonarqube/blob/5ed20116d427b617b04e93bb4f2937ad6692803a/sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java#L806-L863).